cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -310,37 +298,88 @@ def upload_splice_junctions_bed_to_scout(self, dry_run: bool, case_id: str) -> N
+        all_successful: bool = True
+        all_failed: bool = True
+        file_name: str = (
+            "splice junctions bed file" if is_splice_junctions else "bigwig coverage file"
+        )
+        for dna_case_id in dna_cases:
+            if self.status_db.get_case_by_internal_id(internal_id=dna_case_id).is_uploaded:
+                all_failed = False
+                LOG.info(
+                    f"Uploading {file_name} for sample {dna_sample_id} in case {dna_case_id} in Scout."
+                )
+
+                if dry_run:
+                    continue
+                if is_splice_junctions:
                     self.scout_api.upload_splice_junctions_bed(
-                        file_path=splice_junctions_bed.full_path,
+                        file_path=file.full_path,
                         case_id=dna_case_id,
                         customer_sample_id=dna_sample_id,
                     )
-                    LOG.info(
-                        f"Uploaded splice junctions bed file {dna_sample_id} in case {dna_case_id}"
-                    )
                 else:
-                    LOG.warning(
-                        f"Upload of splice junctions bed file for {dna_sample_id} in case {dna_case_id} skipped - "
-                        f"case has not finished uploading"
+                    self.scout_api.upload_rna_coverage_bigwig(
+                        file_path=file.full_path,
+                        case_id=dna_case_id,
+                        customer_sample_id=dna_sample_id,
                     )
+                LOG.info(f"Uploaded {file_name} {dna_sample_id} in case {dna_case_id}")
+            else:
+                all_successful = False
+                LOG.warning(
+                    f"Upload of {file_name} for {dna_sample_id} in case {dna_case_id} skipped - "
+                    f"case has not finished uploading"
+                )
+        return self._upload_success_statement(
+            rna_sample_id=rna_sample_id,
+            all_successful=all_successful,
+            all_failed=all_failed,
+            file_name=file_name,
+        )
 
-        LOG.info("Upload splice junctions bed file finished!")
+    @staticmethod
+    def _upload_success_statement(
+        rna_sample_id: str, all_successful: bool, all_failed: bool, file_name: str
+    ) -> str:
+        """Returns string indicating if all, some or no uploads of splice junctions or bigwig coverage succeeded."""
+        if all_successful and all_failed:
+            return f"{rna_sample_id} - found no related uploaded DNA cases."
+        elif all_successful:
+            return f"{rna_sample_id} - all {file_name} uploads to related uploaded DNA cases successful."
+        elif all_failed:
+            return f"{rna_sample_id} - no {file_name} uploads to related uploaded DNA cases successful."
+        else:
+            return f"{rna_sample_id} - some {file_name} uploads to related uploaded DNA cases successful."
```

Comment:
Maybe it would be better to have a function that precisely tells you which uploads failed or succeeded? Then we don't need this None, Some, All logic?

Reply:
That is indeed a better way to go.